docs/architecture/psa-migration/strategy.md

```diff
@@ -0,0 +1,256 @@
+This document explains the strategy that was used so far in starting the
+migration to PSA Crypto and mentions future perspectives and open questions.
+
+Goals
+=====
+
+Several benefits are expected from migrating to PSA Crypto:
+
+G1. Take advantage of the PSA Crypto driver interface.
+G2. Allow isolation of long-term secrets (for example, private keys).
+G3. Allow isolation of short-term secrets (for example, TLS sesssion keys).
+G4. Have a clean, unified API for Crypto (retire the legacy API).
+
+Currently, some parts of (G1) and (G2) are implemented when
+`MBEDTLS_USE_PSA_CRYPTO` is enabled. For (G2) to take effect, the application
+needs to be changed to use new APIs.
+
+Generally speaking, the numbering above doesn't mean that each goal requires
+the preceding ones to be completed - for example it would be possible to
+start or even complete (G4) before (G3) is even started. However, (G2) and (G3)
+require operations to be done via the PSA Crypto API, which is mostly what (G1)
+is about. Also, we can't retire the legacy API (G4) until we no longer rely on
+it, which again is mostly (G1).
+
+So, a solid intermediate goal would be to complete (G1) when
+`MBEDTLS_USA_PSA_CRYPTO` is enabled - that is, all crypto operations in X.509
+and TLS would be done via the PSA Crypto API.
+
+Compile-time options
+====================
+
+We currently have two compile-time options that are relevant to the migration:
+
+- `MBEDTLS_PSA_CRYPTO_C` - enabled by default, controls the presence of the PSA
+  Crypto APIs.
+- `MBEDTLS_USE_PSA_CRYPTO` - disabled by default (enabled in "full" config),
+  controls usage of PSA Crypto APIs to perform operations in X.509 and TLS
+(G1 above), as well as the availability of some new APIs (G2 above).
+
+The reason why `MBEDTLS_USE_PSA_CRYPTO` is optional, and disabled by default,
+is mostly to avoid introducing a hard (or even default) dependency of X509 and
+TLS and `MBEDTLS_PSA_CRYPTO_C`. This is mostly reasons of code size, and
+historically concerns about the maturity of the PSA code (which we might want
+to re-evaluate).
```

Comment:
Also because it's incompatible with some features (e.g. `MBEDTLS_PSA_CRYPTO_CONFIG`, `MBEDTLS_ECP_RESTARTABLE`).